go/libkb/upak_loader.go

```diff
@@ -289,7 +289,23 @@ func (u *CachedUPAKLoader) loadWithInfo(arg LoadUserArg, info *CachedUserLoadInf
 		if info != nil {
 			info.LoadedLeaf = true
 		}
-		if leaf.public != nil && leaf.public.Seqno == Seqno(upak.Base.Uvv.SigChain) {
+
+		if leaf.eldest == "" {
+			g.Log.CDebugf(ctx, "%s: cache-hit; but user is nuked, evicting", culDebug(arg.UID))
+
+			// Our cached user turned out to be in reset state (without
+			// current sigchain), remove from cache, and then fall
+			// through. LoadUser shall return an error, which we will
+			// return to the caller.
+			u.Lock()
```

Comment:
Is this locking good? I couldn't just call `Invalidate` here because these aren't recursive lock so it's 100% deadlocks then.